huntsman/camera/pyro.py

```diff
@@ -241,16 +247,18 @@ def take_exposure(self,
         
         self.logger.debug(f'Taking {seconds} second exposure on {self.name}: {base_name} in {dir_name}')
         
-        # Remote method call to start the exposure
+        #Remote method call to start the exposure
+        #Once this finishes, the filename is returned
         exposure_result = self._proxy.take_exposure(seconds=seconds,
                                                     base_name=base_name,
                                                     dir_name=dir_name,
                                                     dark=bool(dark),
                                                     *args,
-                                                    **kwargs) 
-        exposure_result
+                                                    **kwargs
+                    #Once exposure is completed, push file to NGAS if necessary
+                    ).then(self._NGASpush, filename_ngas, ngas_push) 
```

Comment:
This is too early to be pushing the file to NGAS. For an image that is taken via `take_observation()` the call to `take_exposure()` will be followed by a call to `process_exposure()`, which in turn calls `_process_fits()`, which adds the info about the `Observation` to the FITS headers with a call to `pocs.utils.fits.update_observation_headers` (at the moment that's all it does). If you push to NGAS here then the ingested files will be missing a lot of important metadata.

The best way to do this is probably to override `_process_fits`, calling the base class version then doing the NGAS push. That would limit NGAS pushes to only exposures started with `take_observation`, but I don't think that's a major problem.